### PR TITLE
BUG: Update CTKAppLauncher from 0.1.30 to 31 to support Qt5 reserved arguments

### PR DESCRIPTION
**Pull request by jcfr**
This commit ensures arguments like `-widgetcount`, `-dpiawareness`, ...
can be passed when executing Slicer.

See https://doc.qt.io/qt-5/qapplication.html#QApplication
and https://doc.qt.io/qt-5/qguiapplication.html#QGuiApplication

List of CTKAppLauncherLib & CTKAPPLAUNCHER changes:

```
$ git shortlog 7e87d4cf5..8759e0398 --no-merges
Jean-Christophe Fillion-Robin (5):
      Begin post-v0.1.30 development [ci skip]
      Main.cpp: Add support for Qt5 built-in command-line arguments
      style: Refactor App4Test introducing display_arguments convenience function
      BUG: Ensure Qt reserved arguments are passed to the launched application
      CTKAppLauncher v0.1.31
```

### CODE REVIEW

**Review by lassoan — APPROVED** [review]
It looks good!

Please add some information about these command-line arguments to the documentation - it could be a new section in https://slicer.readthedocs.io/en/latest/user_guide/settings.html#information-for-advanced-users

No need to describe all the options, but maybe 1-2 commonly used examples and link to Qt documentation, maybe also to the CTK file where all the options are listed.

File: Docs/user_guide/settings.md [hunk 1]
```diff
@@ -142,3 +142,22 @@ The following environment variables can be set before the application is started
 - `SLICER_EXTENSIONS_MANAGER_SERVER_URL`: URL of the extensions manager backend with the `/api` path. Default value is retrieved from the settings using the key `Extensions/ServerUrl`.
 - `SLICER_EXTENSIONS_MANAGER_FRONTEND_SERVER_URL`: URL of the extension manager frontend displaying the web page. Default value is retrieved from the settings using the key `Extensions/FrontendServerUrl`.
 - `SLICER_EXTENSIONS_MANAGER_SERVER_API`: Supported values are `Midas_v1` and `Girder_v1`. Default value is hard-coded to `Girder_v1`.
+
+### Qt built-in command-line options
+
+Slicer application supports specific command-line options that allow modifying the way Qt will interact with the windowing system.
+
+Examples of options:
+- `-widgetcount`, prints debug message at the end about number of widgets left undestroyed and maximum number of widgets existed at the same time.
```
Comment (lassoan) [inline Docs/user_guide/settings.md:151]: Are there any better examples of options that could be actually useful? For example, controlling DPI awareness could be useful. Or setting the window geometry or display number.

File: Docs/user_guide/settings.md [hunk 2]
```diff
@@ -142,3 +142,22 @@ The following environment variables can be set before the application is started
 - `SLICER_EXTENSIONS_MANAGER_SERVER_URL`: URL of the extensions manager backend with the `/api` path. Default value is retrieved from the settings using the key `Extensions/ServerUrl`.
 - `SLICER_EXTENSIONS_MANAGER_FRONTEND_SERVER_URL`: URL of the extension manager frontend displaying the web page. Default value is retrieved from the settings using the key `Extensions/FrontendServerUrl`.
 - `SLICER_EXTENSIONS_MANAGER_SERVER_API`: Supported values are `Midas_v1` and `Girder_v1`. Default value is hard-coded to `Girder_v1`.
+
+### Qt built-in command-line options
+
+Slicer application supports specific command-line options that allow modifying the way Qt will interact with the windowing system.
```
Comment (lassoan) [inline Docs/user_guide/settings.md:148]:
```suggestion
Slicer application accepts standard Qt command-line arguments that specify how Qt interacts with the windowing system.
```

**Review by lassoan — COMMENTED** [review]
Thank you, this is very useful, added a few comments.

File: Docs/user_guide/settings.md [hunk 3]
```diff
@@ -142,3 +142,22 @@ The following environment variables can be set before the application is started
...
+- `-widgetcount`, prints debug message at the end about number of widgets left undestroyed and maximum number of widgets existed at the same time.
+- `-reverse`, sets the application's layout direction to `Qt::RightToLeft`.
+
+To learn about the supported options:
+- https://doc.qt.io/qt-5/qapplication.html#QApplication
+- https://doc.qt.io/qt-5/qguiapplication.html#supported-command-line-options
+
+:::{note}
+
+Since the Slicer launcher is itself a Qt application and the Qt built-in command-line options are expected to be passed to
```
[the remark this answers is not in the document]
Comment (jcfr) [reply Docs/user_guide/settings.md]: I clarified further

File: Docs/user_guide/settings.md [hunk 4]
```diff
@@ -142,3 +142,25 @@ The following environment variables can be set before the application is started
...
 - `SLICER_EXTENSIONS_MANAGER_SERVER_API`: Supported values are `Midas_v1` and `Girder_v1`. Default value is hard-coded to `Girder_v1`.
+
+### Qt built-in command-line options
+
+Slicer application accepts standard Qt command-line arguments that specify how Qt interacts with the windowing system.
+
+Examples of options:
+- `-qwindowgeometry geometry`, specifies window geometry for the main window using the X11-syntax. For example: `-qwindowgeometry 100x100+50+50`.
+- `-display hostname:screen_number`, switches displays on X11 and overrides the `DISPLAY` environment variable.
+- `-platform windows:dpiawareness=[0|1|2]`, sets the [DPI awareness](https://doc.qt.io/qt-5/highdpi.html#microsoft-windows) on Windows.
```
Comment (jcfr) [inline Docs/user_guide/settings.md:153]: I will have to confirm that this syntax works as expected. Will get back to this a few hours ...

Comment (jcfr) [reply Docs/user_guide/settings.md:153]: > I will have to confirm that this syntax works as expected. Will get back to this a few hours ...

:heavy_check_mark: See https://github.com/commontk/AppLauncher/pull/126

Comment (lassoan) [reply Docs/user_guide/settings.md:153]: These are good examples. Hopefully they work!

**Pull request merged**